### PR TITLE
cmds: handle lf as newline character

### PR DESCRIPTION
**Pull request by damianloew**
## Motivation and Context

 - JIRA: PD-308

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Comment by agkaminski** [comment]
This might turn out problematic - some terminals are sending only CR, some only LF, but there are some (mostly Windows based) that are sending CRLF. So this solution will cause empty new line to be interpreted on every enter on such terminals.

What we can do: e.g. on CR save a flag, that CR has been read and if next char is LF ignore it.

**Comment by damianloew** [comment]
> This might turn out problematic - some terminals are sending only CR, some only LF, but there are some (mostly Windows based) that are sending CRLF. So this solution will cause empty new line to be interpreted on every enter on such terminals.
> 
> What we can do: e.g. on CR save a flag, that CR has been read and if next char is LF ignore it.

Please check the solution with the `newline` flag and let me know whether it's ok

File: cmds/cmd.c [hunk 1]
```diff
@@ -196,7 +197,15 @@ void cmd_prompt(void)
 
 		/* Regular characters */
 		if (c) {
-			if (c == '\r') {
+			if ((c == '\r') || (c == '\n')) {
+				if (newline) {
```
Comment (agkaminski) [inline cmds/cmd.c:201]:
```suggestion
				if (newline != 0) {
```

**Review by agkaminski — COMMENTED** [review]
We should ignore only LR if CR was previously parsed - right now e.g. double enter won't work (we expect two empty lines, we'll get one) ;)

File: cmds/cmd.c (hunk 1 above)
Comment (damianloew) [reply cmds/cmd.c]: You are right, sorry for that. The last force-push handles this case.

File: cmds/cmd.c [hunk 2]
```diff
@@ -196,7 +197,17 @@ void cmd_prompt(void)
 
 		/* Regular characters */
 		if (c) {
-			if (c == '\r') {
+			if ((c == '\r') || (c == '\n')) {
+				/* handle crlf line ending */
+				if ((c == '\r') && (newline == 0)) {
```
Comment (agkaminski) [inline cmds/cmd.c:202]: `newline == 0` is redundant - we always end up with `newline` = 1 anyway
```suggestion
				if (c == '\r') {
```

**Review by agkaminski — APPROVED** [review]
One nitpick, otherwise LGTM

File: cmds/cmd.c (hunk 2 above)
Comment (damianloew) [reply cmds/cmd.c]: Yeah, maybe that's better, changed.

**Review by agkaminski — APPROVED** [review]

**Pull request merged**